picard/ui/searchdialog.py

```diff
@@ -0,0 +1,362 @@
+                reverse=True,
+                key=itemgetter(0))
+            tracks = [item[3] for item in tmp]
+
+        del self.search_results[:]  # Clear existing data
+        self.parse_tracks_from_xml(tracks)
+        self.display_results()
+
+    def display_results(self):
+        self.show_table(self.table_headers)
+        for row, obj in enumerate(self.search_results):
+            track = obj[0]
+            table_item = QtGui.QTableWidgetItem
+            self.table.insertRow(row)
+            self.table.setItem(row, 0, table_item(track.title))
+            self.table.setItem(row, 1, table_item(track.length))
+            self.table.setItem(row, 2, table_item(track.artist))
+            self.table.setItem(row, 3, table_item(track.release))
+            self.table.setItem(row, 4, table_item(track.date))
+            self.table.setItem(row, 5, table_item(track.country))
+            self.table.setItem(row, 6, table_item(track.release_type))
+
+    def parse_tracks_from_xml(self, tracks_xml):
+        for node in tracks_xml:
+            rec_id = node.id
+            rec_title = node.title[0].text
+            artist = artist_credit_from_node(node.artist_credit[0])[0]
+            try:
+                length = format_time(node.length[0].text)
+            except AttributeError:
+                length = ""
+            try:
+                releases = node.release_list[0].release
+                for release in releases:
+                    rel_id = release.id
+                    rel_title = release.title[0].text
+                    if "date" in release.children:
+                        date = release.date[0].text
+                    else:
+                        date = None
+                    if "country" in release.children:
+                        country = release.country[0].text
+                    else:
+                        country = ""
+                    rg = release.release_group[0]
+                    rg_id = rg.id
+                    types_list = []
+                    if "primary_type" in rg.children:
+                        types_list.append(rg.primary_type[0].text)
+                    if "secondary_type_list" in rg.children:
+                        for sec in rg.secondary_type_list:
+                            types_list.append(sec.secondary_type[0].text)
+                    types = "+".join(types_list)
+
+                    track = Track(
+                            id=rec_id,
+                            release_id=rel_id,
+                            rg_id=rg_id,
+                            title=rec_title,
+                            artist=artist,
+                            length=length,
+                            release=rel_title,
+                            date=date,
+                            country=country,
+                            release_type=types)
+                    self.search_results.append((track, node))
+
+            except AttributeError:
```

Comment:
Handling this via exception makes the control flow harder to follow (where is the exception thrown?) and seems strange since we know when it should happen (`not (hasattr(node, 'release_list') and node.release_list)` right?).